hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/AbstractEntityInitializer.java

```diff
@@ -397,7 +397,12 @@ public void resolveInstance(RowProcessingState rowProcessingState) {
 					rowProcessingState.getJdbcValuesSourceProcessingState(),
 					this
 			);
-			isOwningInitializer = holder.getEntityInitializer() == this;
+			if ( holder.getEntity() == null ) {
+				isOwningInitializer = true;
+			}
+			else {
+				isOwningInitializer = holder.getEntityInitializer() == this;
+			}
```

Comment:
Just a minor suggestion. I am curious though, is the result of `getEntity()` the enhanced proxy?

```suggestion
			isOwningInitializer = holder.getEntity() == null || holder.getEntityInitializer() == this;
```